### PR TITLE
Prohibit detail categories from having hierarchical division

### PR DESCRIPTION
**Pull request by nechaido**

### CODE REVIEW

File: lib/metaschema-config/category.js [hunk 1]
```diff
@@ -58,6 +58,16 @@ const addCategoryData = (type, entity, ms) => {
   return errors;
 };
 
+const checkDetail = category =>
+  ['Catelog', 'Subdivision', 'Hierarchy']
```
Comment (lundibundi) [inline lib/metaschema-config/category.js]:
```suggestion
  ['Catalog', 'Subdivision', 'Hierarchy']
```

Comment (lundibundi) [inline lib/metaschema-config/category.js]: Also, why not make them lowercase to begin with?

File: lib/metaschema-config/category.js [hunk 2]
```diff
@@ -58,6 +58,16 @@ const addCategoryData = (type, entity, ms) => {
...
+    .filter(type => category[type.toLowerCase()])
+    .map(
+      type =>
+        new SchemaValidationError('detailHierarchy', category.name, {
```
Comment (lundibundi) [inline lib/metaschema-config/category.js]: Could we add to the error message what category the detail comes from to be clear?
(`Category '${source}' cannot be divided by ${type} while being a detail in '${categoryName}.${fieldName}`)
(`Category '${source}' cannot be divided by ${type} while being a detail (in '${categoryName}.${fieldName})`)

File: lib/metaschema-config/category.js (hunk 1 above)
Comment (nechaido) [reply lib/metaschema-config/category.js]: So that error message says `can not be divided by Catalog` not `can not be divided by catalog`.

**Comment by nechaido** [comment]
ping @belochub.

**Pull request merged**